src/main/java/reposense/ConfigParser/CliArgumentsParser.java

```diff
@@ -0,0 +1,82 @@
+package reposense.ConfigParser;
+
+import reposense.exceptions.ParseException;
+
+import java.util.Date;
+import java.util.HashMap;
+
+public class CliArgumentsParser extends Parser<Argument, String[]> {
+
+    private static final String ARGUMENT_PREFIX_DASH = "-";
+
+    private static final String CONFIG_FILE_ARG = "config";
+    private static final String TARGET_FILE_ARG = "output";
+    private static final String SINCE_DATE_ARG = "since";
+    private static final String UNTIL_DATE_ARG = "until";
+
+    private static final String PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS = "Failed to parse inputs argumentMap";
+    private static final String PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE = "you need to specify a config CSV file!";
+
+    private final HashMap<String, String> argumentMap;
+    private final DateParser dateParser;
+    private final Argument argument;
+
+    public CliArgumentsParser() {
+        argumentMap = new HashMap<String, String>();
+        dateParser = new DateParser();
+        argument = new Argument();
+    }
+
+    @Override
+    public Argument parse(String[] args) throws ParseException{
+        for (int i = 0; i < args.length; i = i + 2) {
+
+            if (i + 1 > args.length) {
+                throw new ParseException(PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS);
+            }
+
+            if (!args[i].startsWith(ARGUMENT_PREFIX_DASH)) {
+                throw new ParseException(PARSE_EXCEPTION_MESSAGE_INVALID_INPUTS);
+            }
+
+            String key = args[i].substring(1);
+            argumentMap.put(key, args[i + 1]);
+        }
+
+        if (isAllMandatoryArgumentsPresent()) {
+            throw new ParseException(PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE);
```

Comment:
`PARSE_EXCEPTION_MESSAGE_NO_CSV_FILE` should be only a subset of `isAllMandatoryArgumentsPresent`